cmd/dockerd/config_unix.go

```diff
@@ -27,7 +27,8 @@ func installConfigFlags(conf *config.Config, flags *pflag.FlagSet) {
 	flags.Var(opts.NewNamedUlimitOpt("default-ulimits", &conf.Ulimits), "default-ulimit", "Default ulimits for containers")
 	flags.BoolVar(&conf.BridgeConfig.EnableIPTables, "iptables", true, "Enable addition of iptables rules")
 	flags.BoolVar(&conf.BridgeConfig.EnableIP6Tables, "ip6tables", true, "Enable addition of ip6tables rules")
-	flags.BoolVar(&conf.BridgeConfig.EnableIPForward, "ip-forward", true, "Enable net.ipv4.ip_forward")
+	flags.BoolVar(&conf.BridgeConfig.EnableIPForward, "ip-forward", true, "Enable IP forwarding in system configuration")
+	flags.BoolVar(&conf.BridgeConfig.EnableFilterForwardDrop, "ip-filter-forward-drop", true, "Set the filter-FORWARD policy to DROP when enabling IP forwarding")
```

Comment:
This needs an update in [man/dockerd.8.md](https://github.com/moby/moby/blob/6ac445c42badc98b416baa32cca5d6f8b712b359/man/dockerd.8.md) and the corresponding daemon reference docs in [docker/cli](https://github.com/docker/cli/blob/2995631498fc00463e387bf60e45947db5c6e025/docs/reference/dockerd.md)

Also curious; should enabling this without the other option be an error-condition? Or is it fine to ignore?

Reply:
Added to the manpage.

I think it's ok to ignore - it says it'll only do-something if the daemon enables forwarding itself, and it won't if `--ip-forward=false` (or if iptables/ip6tables is disabled).

Reply:
Oh, I didn't look at this closely; I just realised we're introducing a new boolean flag, but with `true` as default. I realise we already have some those, but if we consider this default to be `true` in the foreseeable future, then perhaps we should have `false` as default and name the flag accordingly.

Having `false` makes the UX more usable (instead of `--boolean-flag=false`), but also can help prevent issues when reading config from `daemon.json` where a missing field would default to "false" when unmarshaling.


(just a quick comment; about to jump in a call)

Reply:
Right, yes - the closely related `--iptables`, `--ip6tables` and `--ip-forward` flags all default to `true` ... I guessed that was probably deliberate! They're all "batteries included" flags ... you need to leave them enabled to make things work, unless you know about the innards and have a good reason to turn them off.

So, I'm not sure a default-false option fits well with those others and, if we do change it, it'd need to be called something like `--no-ip-filter-forward-drop`. Then it's a bit too double-negative-y, "`--no-ip-filter-forward-drop=false`" means it will set the filter-FORWARD policy to DROP if it needs to.

(Unless anyone has a better idea for the flag name, it is already a bit cumbersome ... but I thought it was good to start it with "`--ip...`" so that it's close to those other flags in help text, and I think it'll still make sense for native nftables.)

I don't think there's an issue with merging CLI and daemon.json options, in the same way as for the other options - no option is allowed to be specified in both, if it's in the JSON it'll override the CLI's default. If not present in the JSON or the CLI, the CLI's default wins.

Reply:
After further discussion in Slack, I've changed the option to `--ip-forward-no-drop`.

No engineers were harmed in the making of this decision!